fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1299,17 +1289,24 @@ private SqlQueryData processMissingParm(Class<?> resourceType, Parameter queryPa
 
         }
 
+        // Build this piece of the segment
+        // missing: LEFT JOIN (SELECT DISTINCT LOGICAL_RESOURCE_ID FROM ...)
+        //            TEMP ON TEMP.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID
+        //            WHERE TEMP.LOGICAL_RESOURCE_ID is NULL
+        // not missing: JOIN (SELECT DISTINCT LOGICAL_RESOURCE_ID FROM ...)
+        //            TEMP ON TEMP.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID            
         if (missing == null || missing) {
-            whereClauseSegment.append("NOT ");
+            whereClauseSegment.append(" LEFT JOIN ");
+        } else {
+            whereClauseSegment.append(" JOIN ");
         }
-        whereClauseSegment.append("EXISTS ").append("(SELECT 1 FROM " + valuesTable + WHERE);
-
-        // Build this piece of the segment:
-        // (P1.PARAMETER_NAME_ID = x AND P1.logical_resource_id = R.logical_resource_id))
+        whereClauseSegment.append("(SELECT DISTINCT LOGICAL_RESOURCE_ID FROM " + valuesTable + WHERE);
         this.populateNameIdSubSegment(whereClauseSegment, queryParm.getCode(), valuesTable.toString());
-        whereClauseSegment.append(AND).append(valuesTable + DOT + "LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID");
         whereClauseSegment.append(RIGHT_PAREN).append(RIGHT_PAREN);
-        // }
+        whereClauseSegment.append(" TEMP ON TEMP.LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID");
```

Comment:
I think using another name here is cleaner. 
For instance, VALUES_TABLE that way the code is easier to say ah... it's the values table.

Reply:
there is only logical_resource_id in it, so It didn't name it as Values_table to avoid confusing, but I will add Observation_STR_VALUES into the comments to make it clearer.